protocols/imap/src/main/java/org/apache/james/imap/encode/AnnotationResponseEncoder.java

```diff
@@ -0,0 +1,72 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.encode;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.james.imap.api.ImapConstants;
+import org.apache.james.imap.api.ImapMessage;
+import org.apache.james.imap.api.process.ImapSession;
+import org.apache.james.imap.encode.base.AbstractChainedImapEncoder;
+import org.apache.james.imap.message.response.AnnotationResponse;
+import org.apache.james.mailbox.model.MailboxAnnotation;
+
+public class AnnotationResponseEncoder extends AbstractChainedImapEncoder {
+
+    public AnnotationResponseEncoder(ImapEncoder next) {
+        super(next);
+    }
+
+    protected void doEncode(ImapMessage acceptableMessage, final ImapResponseComposer composer, ImapSession session) throws IOException {
+
+        AnnotationResponse response = (AnnotationResponse) acceptableMessage;
+
+        composer.untagged();
+        composer.commandName(ImapConstants.ANNOTATION_RESPONSE_NAME);
+
+        composeMailbox(composer, response.getMailboxName());
+        composeAnnotations(composer, response.getMailboxAnnotations());
+
+        composer.end();
+    }
+
+    private void composeMailbox(ImapResponseComposer composer, String mailboxName) throws IOException {
+        composer.quote(mailboxName == null ? "" : mailboxName);
```

Comment:
It is not a mistake, your code works :-) . 

It is just something hard to read so we prefer not to use it, and Optionals allow also to write it on one line and are readble ;-)